src/ui/find-ui/src/FamilyHubs.ServiceDirectory.Web/Pages/ServiceDetail/Index.cshtml.cs

```diff
@@ -0,0 +1,159 @@
+                Address = serviceLocation.GetAddress(),
+                Accessibilities = GetAccessibilities(serviceLocation.AccessibilityForDisabilities)
+            });
+
+        return locations;
+    }
+
+    private static ScheduleDto? GetScheduleForLocation(long locationId,
+        ICollection<ServiceAtLocationDto> serviceAtLocations)
+        => serviceAtLocations.FirstOrDefault(x => x.LocationId == locationId)?.Schedules.FirstOrDefault();
+
+    private static string IsFamilyHub(LocationTypeCategory locationTypeCategory) =>
+        locationTypeCategory == LocationTypeCategory.FamilyHub ? "Yes" : "No";
+
+    private static IEnumerable<string> GetAccessibilities(
+        ICollection<AccessibilityForDisabilitiesDto> accessibilityForDisabilities)
+        => [..accessibilityForDisabilities.Select(x => x.Accessibility).OfType<string>()];
+
+    private static Contact GetContact(ICollection<ContactDto> serviceContacts)
+    {
+        ContactDto? contactDto = serviceContacts.FirstOrDefault();
+
+        return new Contact
+        {
+            Email = contactDto?.Email,
+            Phone = contactDto?.Telephone,
+            TextMessage = contactDto?.TextPhone,
+            Website = contactDto?.Url
+        };
+    }
+
+    private static ServiceDetailModel GetServiceDetailModel(ServiceDto serviceDto)
+    {
+        ScheduleDto? serviceScheduleDto =
+            serviceDto.Schedules.FirstOrDefault(x => x.ServiceId == serviceDto.Id);
+
+        string attendingTypes = GetAttendingTypes(GetDeliveries(serviceDto.ServiceDeliveries));
+        string onlineTelephoneHeader = GetOnlineTelephoneHeader(attendingTypes);
+
+        ServiceDetailModel serviceDetailModel = new()
+        {
+            Name = serviceDto.Name,
+            Summary = serviceDto.Summary,
+            Eligibility = GetEligibility(serviceDto.Eligibilities),
+            Cost = GetCostOption(serviceDto.CostOptions),
+            MoreDetails = serviceDto.Description,
+            AttendingTypes = attendingTypes,
+
+            OnlineTelephone = onlineTelephoneHeader,
+            Schedule = !string.IsNullOrWhiteSpace(onlineTelephoneHeader)
+                ? new Schedule
+                {
+                    DaysAvailable = GetDaysAvailable(serviceScheduleDto) ?? "None provided",
+                    ExtraAvailabilityDetails = serviceScheduleDto?.Description ?? "None provided",
+                }
+                : null,
+
+            Categories = serviceDto.Taxonomies.Select(t => t.Name).Order(),
+            Languages = serviceDto.Languages.Select(l => l.Name).Order(),
+
+            Locations = GetLocations(serviceDto.Locations, serviceDto.ServiceAtLocations),
+            Contact = GetContact(serviceDto.Contacts)
+        };
+
+        return serviceDetailModel;
+    }
+
+    public async Task<IActionResult> OnGetAsync(long serviceId, string fromUrl)
+    {
+        ServiceDto serviceDto = await GetService(serviceId);
+
+        BackUrl = fromUrl;
```

Comment:
We should validate that url isn't malicious. Probably just whitelist to make sure it's going to the search page?

Reply:
Cheers for suggesting `GetEncodedPathAndQuery`!!

Resolved: [b2c9fe5](https://github.com/DFE-Digital/fh-services/pull/206/commits/b2c9fe58e54dd1b3ace932ec493bb6ee60735207)